### DISCUSSION TITLE
Allow to change NGINX root directory

### DISCUSSION THREAD

**Discussion by rsov** [serversideup/docker-php#224] [Ideas & Feature Requests]
# 👉 Describe the problem

Magento 2 want's the root to be `/var/www/pub` while the configured one is `/var/www/public`

# 👥 Problem evidence & reach

Searched the Magento 2 source and it's hardcoded everywhere, see their nginx config sample: https://github.com/magento/magento2/blob/2.4-develop/nginx.conf.sample

# 🥰 Describe the "impact" on users?

Anyone who want's to run mage 2 without configuring much. There are dozens of us

# 🏆 How to solve this problem

Allow change by ENV variable 

(Currently have a symlink made to solve this)

**Comment by jaydrogers** [comment]
You may want to look into doing this with the new beta: https://github.com/serversideup/docker-php/discussions/254

Look at the `APP_BASE_DIR` variable: https://release-v3-0.docker-php.pages.dev/docs/reference/environment-variable-specification

**Reply by rsov** [reply]
🙌  Legend!
